src/pk_gen.rs

```diff
@@ -208,7 +208,16 @@ pub(crate) fn pk_gen(
     pivots: &mut u64,
 ) -> i32 {
     let mut buf = [0u64; 1 << GFBITS];
-    let mut mat = [[0u8; SYS_N / 8]; PK_NROWS];
+    let mut mat;
+    #[cfg(feature = "alloc")]
+    {
+        use alloc::boxed::Box;
+        mat = Box::new([[0u8; SYS_N / 8]; PK_NROWS]);
```

Comment:
It would be really good if any allocation done *inside* the library itself would also respect the `zeroize` feature. If that feature is enabled it would be nice if the library does not spill parts of secrets into heap memory IMOH

Reply:
What I meant was not to avoid the box. But to make sure it is zeroized before deallocated :)

Thanks for working on this! :heart:

Reply:
Good catch! Thanks for mentioning.

@faern does it make sense to introduce zeroize here? I mean the implementation is the *public* key generation and the values do not seem to touch `sk` in any way. It is however the only allocation introduced in the cryptographic routines. Thus I am very glad we discuss it and pay special attention.

My understanding is that I have overlooked `zeroize` for the KATKEM tests. Because `Testcase` structs contain the secret key and `Testcase` structs may now be heap-allocated.

Reply:
hey @faern thank you very much for reviewing :) I'll try to respect the `zeroize` feature for all our Box:: heap allocations aswell.

As a first step this meant to change the PublicKey type to `KeyBufferMut` and implementing the zeroize + zeroizeonDrop traits for it. I added the assertion of the public key to the zeroize tests aswell as can be seen in the latest commit

If this is deemed a wrong approach please inform me on that